qoffer-infrastructure/src/main/groovy/life/qbic/portal/qoffer2/web/views/create/offer/CustomerSelectionView.groovy

```diff
@@ -185,4 +198,17 @@ class CustomerSelectionView extends VerticalLayout{
             next.setEnabled(true)
         })
     }
+
+    private void addFilters(ListDataProvider customerListDataProvider) {
+        HeaderRow customerFilterRow = customerGrid.appendHeaderRow()
+        GridUtils.setupColumnFilter(customerListDataProvider,
+                customerGrid.getColumn("FirstName"),
+                customerFilterRow)
+        GridUtils.setupColumnFilter(customerListDataProvider,
+                customerGrid.getColumn("LastName"),
+                customerFilterRow)
+        GridUtils.setupColumnFilter(customerListDataProvider,
+                customerGrid.getColumn("EmailAddress"),
+                customerFilterRow)
+    }
```

Comment:
I would keep the filter setup with the corresponding column setup. This way the setup for one column is done in place. Also you do not need to set the Id as you get the column as return value and can just use it. (you only need to modify it at setup and not later on)

Reply:
I adhered to the SRP for methods and decoupled the setup of the column and the filter generation. I do not see a reasonable disadvantage doing it like that.